IHP/Pagination/ViewFunctions.hs

```diff
@@ -0,0 +1,136 @@
+module IHP.Pagination.ViewFunctions (
+    module IHP.Pagination.Types,
+    renderPagination,
+    renderFilter,
+) where
+
+import IHP.Prelude
+import IHP.Pagination.Types
+import IHP.Pagination.Helpers
+import IHP.Pagination.ControllerFunctions
+
+import IHP.ControllerSupport
+
+import Text.Blaze.Html (Html)
+import IHP.HSX.QQ (hsx)
+
+import IHP.Controller.Param (paramOrNothing)
+
+
+-- | Render a navigation for your pagination. This is to be used in your view whenever 
+-- to allow users to change pages, including "Next" and "Previous".
+renderPagination :: (?context::ControllerContext) => Pagination -> Html
+renderPagination pagination@Pagination {currentPage, window, pageSize} =
+    [hsx|
+        <nav aria-label="Page Navigator">
+            <ul class="pagination justify-content-center">
+                <li class={prevClass}>
+                    <a class="page-link" href={pageUrl $ currentPage - 1} aria-label="Previous">
+                        <span aria-hidden="true">&laquo;</span>
+                        <span class="sr-only">Previous</span>
+                    </a>
+                </li>
+                {renderItems}
+                <li class={nextClass}>
+                    <a class="page-link" href={pageUrl $ currentPage + 1} aria-label="Previous">
+                        <span aria-hidden="true">&raquo;</span>
+                        <span class="sr-only">Next</span>
+                    </a>
+                </li>
+            </ul>
+        </nav>
+    |]
+        where
+            nextClass = "page-item" ++ (if hasNextPage pagination then "" else " disabled") :: Text
```

Comment:
IHP has a `classes` helper for this :)

```suggestion
            nextClass = classes ["page-item", (hasNextPage pagination, "disabled")]
```

Reply:
Forgive my potential ignorance, is the `(Bool,a)` syntax for including an element in a list correct? Is it provided by a language extension?

Reply:
[This is provided by the `IHP.View.Classes` module](https://ihp.digitallyinduced.com/api-docs/IHP-View-Classes.html) :) The implementation is relatively short:

```haskell
classes :: [(Text, Bool)] -> Text
classes !classNameBoolPairs =
    classNameBoolPairs
    |> filter snd
    |> map fst
    |> unwords
```

It's initially inspired by react's classnames helper, which is now replaced by https://www.npmjs.com/package/classnames

Reply:
Oh, okay, I see. But the list above still confuses me because it appears to have elements of different types. Perhaps just a typo? Would it be something like `[("page-item", True), ("disabled", hasNextPage pagination)]`?

Reply:
Got it :) The `"page-item"` is a trick. We have a `IsString` instance for this in `IHP.View.Classes`.

The `["page-item", ("disabled", hasNextPage pagination)]` is rewritten to `[fromString "page-item", (fromString "disabled", hasNextPage pagination)]` because of the GHC `OverloadedStrings` extension.

Then we have a special `IsString` instance in `IHP.View.Classes`:

```haskell
instance IsString (Text, Bool) where
    fromString string = (cs string, True)
```

Now if you substitute the `fromString "page-item"` with the `(cs "page-item", True)` you have matching types :)

TLDR: `"page-item"` can be the short form of `("page-item", True)` if the `IsString (Text, Bool)` type instance  is used.

Reply:
Ahh yes `IsString` and `OverloadedStrings`!!! Of course!!!!! I should have known. Very clever 😃 You guys really make clever use of GHC. 😊 (I know this one isn't particularly fancy, but there are lots of other clever ways you use extensions too!)

Very cool, thanks.

Reply:
Sidenote, I sometimes find it a bit hard to know which extensions are enabled since you hide that from the source file itself. Which I agree with for most (non-Haskell) users since you don't want to overwhelm them with details. But it makes it hard to track down these sorts of things and what actually enables them.

Reply:
:) agree, we could add a list of all enabled language extensions to [`Helpful Tips & Special Haskell Syntax`](https://ihp.digitallyinduced.com/Guide/helpful-tips.html). Do you think this would improve the situation?

Reply:
Frankly, I had not found that particular page! We could do that, yes. But I like how that page gives you the syntax and then tells you what enables it (at least for some of them). Either way, it's not like it really affected my experience at the beginning. I just decided to take certain things for granted as I was starting, as if it was a brand new language. But as I got more curious I wanted to know how things actually worked under the hood.